lib/DxcSupport/FileIOHelper.cpp

```diff
@@ -753,6 +755,10 @@ HRESULT DxcCreateBlobEncodingFromBlob(
       return S_OK;
     }
     if (encodingKnown) {
+      if (TryCreateBlobUtfFromBlob(pFromBlob, codePage, pMalloc,
+                                   +ppBlobEncoding)) {
```

Comment:
Separate issue, it's been a while since I've looked at this file. Can you explain why this addition was necessary? It looks as though `TryCreateBlobUtfFromBlob` allows UTF8 or UTF16 encoding, but it's not clear to me how such encodings can be reached from existing calls to this function.

Reply:
Yes stray character thanks for the point out. And second, Tex asked me to add this to my change when he was looking into code paths for blob creation that came up from a conversation around encoded files with BOMs in PIX. From what I understand, the addition here enables the creation of `InternalDxcBlobUtf8`/`InternalDxcBlobUtf16` from within `TryCreateBlobUtfFromBlob` if the buffer is already null terminated. As Tex mentioned, without the fix there would be an extra step if the utf blob interface was requested. I think like with the call directly below to `InternalDxcBlobEncoding::CreateFromBlob` Does this answer your question?